doc/specs/stdlib_selection.md

```diff
@@ -0,0 +1,340 @@
+---
+title: Selection Procedures
+---
+
+# The `stdlib_selection` module
+
+[TOC]
+
+## Overview of selection
+
+Suppose you wish to find the value of the kth-smallest entry in an array of size N, or
+the index of that value. While it could be done by sorting the whole array
+using `[[stdlib_sorting(module):sort(interface)]]` or 
+`[[stdlib_sorting(module):sort_index(interface)]]` from 
+`[[stdlib_sorting(module)]]` and then finding the k-th entry, that would
+require O(N x LOG(N)) time. However selection of a single entry can be done in
+O(N) time, which is much faster for large arrays.  This is useful, for example,
+to quickly find the median of an array, or some other percentile.
+
+The Fortran Standard Library therefore provides a module, `stdlib_selection`,
+which implements selection algorithms.
+
+## Overview of the module
+
+The module `stdlib_selection` defines two generic subroutines:
+* `select` is used to find the kth-smallest entry of an array. The input
+array is also modified in-place, and on return will be partially sorted
+such that `all(array(1:k) <= array(k)))`  and `all(array(k) <= array((k+1):size(array)))`.
+The user can optionally specify `left` and `right` indices to constrain the search
+for the kth-smallest value. This can be useful if you have previously called `select`
+to find a smaller or larger rank (that will have led to partial sorting of
+`array`, thus implying some constraints on the location).
+
+* `arg_select` is used to find the index of the kth-smallest entry of an array.
+In this case the input array is not modified, but the user must provide an
+input index array with the same size as `array`, having unique indices from
+`1:size(array)`, which is modified instead. On return the index array is modified
+such that `all(array(index(1:k)) <= array(index(k)))` and `all(array(k) <= array(k+1:size(array)))`.
+The user can optionally specify `left` and `right` indices to constrain the search
+for the kth-smallest value. This can be useful if you have previously called `arg_select`
+to find a smaller or larger rank (that will have led to partial sorting of
+`index`, thus implying some constraints on the location).
+
+#### Licensing
+
+The Fortran Standard Library is distributed under the MIT
+License. However components of the library may be based on code with
+additional licensing restrictions. In particular `select` and `arg_select`
```

Comment:
Based on the following sentences, there are no additional restrictions compared to MIT license.
Therefore I propose the following sentences:

```suggestion
License. It is worth noting that  `select` and `arg_select`
```